server/schedule/labeler/labeler.go

```diff
@@ -122,9 +178,26 @@ func (l *RegionLabeler) GetLabelRules(ids []string) ([]*LabelRule, error) {
 
 // GetLabelRule returns the Rule with the same ID.
 func (l *RegionLabeler) GetLabelRule(id string) *LabelRule {
-	l.RLock()
-	defer l.RUnlock()
-	return l.labelRules[id]
+	return l.getAndCheckRule(id, time.Now())
+}
+
+func (l *RegionLabeler) getAndCheckRule(id string, now time.Time) *LabelRule {
+	l.Lock()
+	defer l.Unlock()
+	rule, ok := l.labelRules[id]
+	if !ok {
+		return nil
+	}
+	if !rule.checkAndRemoveExpireLabels(now) {
+		return rule
+	}
+	if len(rule.Labels) == 0 {
+		l.storage.DeleteRule(id)
```

Comment:
Here It will do nothing here and the item wouldn't be deleted in the storage until `restart`.  However, if there is something wrong with the storage last for long, the restart of the server would be long with too many expired items.